src/Widgets/FeedRow.vala

```diff
@@ -44,9 +44,13 @@ public class FeedReader.FeedRow : Gtk.ListBoxRow {
 			var rowhight = 30;
 			m_box = new Gtk.Box(Gtk.Orientation.HORIZONTAL, 0);
 			m_icon = createFavIcon();
-
 			m_icon.margin_start = level * 24;
 
+			FavIconManager.get_default().ReloadFavIcon.connect(feed => {
+				if(m_feed.getFeedID() == feed.getFeedID())
+					reloadFavIcon.begin();
+			});
```

Comment:
Do we need to manually disconnect this signal in the destructor? I'm not sure how Vala handles this